external-reports/src/test/java/org/sonar/java/externalreport/PmdSensorTest.java

```diff
@@ -218,3 +223,66 @@ private static String language(Path file) {
     return path.substring(path.lastIndexOf('.') + 1);
   }
 }
+
+class ExternalIssueAsserter {
```

Comment:
Just a small comment about this: I'm not a big fan of having two classes at the same level in a compilation unit, just one being not public. I would prefer to make it a public static method inside the other one, or a dedicated class in its own compilation unit.

Also, what about making it inherit from AssertJ's `AbstractAssert`?
https://joel-costigliola.github.io/assertj/assertj-core-custom-assertions.html

Reply:
I thought about AssertJ approach. Unfortunately, I am not sure we'll benefit much from it. Cause we still have to implement all needed assertions logic.

I also changed the logic of assertions, so there is only one point to get an assertion error and we'll have all the error messages collected.

Reply:
Fine by me :)

I just believe that the more we stay closer from how AssertJ operate, the higher is the chance to have it reused somewhere in our own tests.